From fd5f3e8f16fe9da9dc5b553144eaf0457847c6fc Mon Sep 17 00:00:00 2001
From: Prateek SU <41370460+prateekmedia@users.noreply.github.com>
Date: Wed, 3 Nov 2021 12:54:31 +0530
Subject: [PATCH] Workaround for 5.23 from Luwx/LightlyShaders#26

---
 plugins/roundedwindow/roundedwindow.cpp | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/plugins/roundedwindow/roundedwindow.cpp b/plugins/roundedwindow/roundedwindow.cpp
index 85379a3..04322a3 100644
--- a/plugins/roundedwindow/roundedwindow.cpp
+++ b/plugins/roundedwindow/roundedwindow.cpp
@@ -117,9 +117,11 @@ static KWin::GLShader *getShader()
     } else if (traits & KWin::ShaderTrait::UniformColor)
         stream << "uniform vec4 geometryColor;\n";
 
+    #if KWIN_EFFECT_API_VERSION < 233
     if (traits & KWin::ShaderTrait::ClampTexture) {
         stream << "uniform vec4 textureClamp;\n";
     }
+    #endif
 
     if (output != QByteArrayLiteral("gl_FragColor"))
         stream << "\nout vec4 " << output << ";\n";
@@ -236,6 +238,7 @@ bool RoundedWindow::enabledByDefault()
     return supported();
 }
 
+#if KWIN_EFFECT_API_VERSION < 233
 bool RoundedWindow::hasShadow(KWin::WindowQuadList &qds)
 {
     for (int i = 0; i < qds.count(); ++i)
@@ -244,6 +247,7 @@ bool RoundedWindow::hasShadow(KWin::WindowQuadList &qds)
 
     return false;
 }
+#endif
 
 bool RoundedWindow::isMaximized(KWin::EffectWindow *w)
 {
@@ -281,7 +285,10 @@ void RoundedWindow::drawWindow(KWin::EffectWindow *w, int mask, const QRegion &r
             || w->isDock()
             || w->isPopupWindow()
             || w->isPopupMenu()
-            || !hasShadow(data.quads)) {
+            #if KWIN_EFFECT_API_VERSION < 233
+                 || !hasShadow(data.quads)
+            #endif
+        ) {
         if (!allowList.contains(w->windowClass()))
             return KWin::Effect::drawWindow(w, mask, region, data);
     }
